runtime/jailer_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"syscall"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/firecracker-microvm/firecracker-containerd/internal"
+)
+
+func TestCopyFile_simple(t *testing.T) {
+	srcPath := "./firecracker-runc-config.json.example"
+	dstPath := "./test-copy-file"
+
+	err := copyFile(srcPath, dstPath, 0600)
+	assert.NoError(t, err, "failed to copy file")
+	defer os.Remove(dstPath)
+
+	_, err = os.Stat(dstPath)
+	assert.NoError(t, err, "failed to stat file")
+}
+
+func TestCopyFile_invalidPaths(t *testing.T) {
+	srcPath := "./invalid.path"
+	dstPath := "./test-copy-file"
+
+	err := copyFile(srcPath, dstPath, 0600)
+	assert.Error(t, err, "copyFile should have returned an error")
+}
+
+func TestChownR(t *testing.T) {
+	internal.RequiresRoot(t)
+
+	testPath, err := ioutil.TempDir("./", "TestChownR")
+	assert.NoError(t, err, "failed to create temp directory")
+	err = os.Chmod(testPath, 0666)
+	assert.NoErrorf(t, err, "failed to chown %v", testPath)
+	err = os.MkdirAll(filepath.Join(testPath, "nested"), os.ModePerm)
+	assert.NoErrorf(t, err, "failed to create %v", testPath)
+
+	defer func() {
+		os.RemoveAll(testPath)
+	}()
+
+	fooPath := filepath.Join(testPath, "foo")
+	err = ioutil.WriteFile(fooPath, []byte{}, os.ModePerm)
```

Comment:
While this is in the temporary directory in a test, os.ModePerm (0777) is unnecessary open.

I think the aim of this constant is for masking other bits (https://github.com/golang/go/search?q=ModePerm&unscoped_q=ModePerm), rather than as an actual permission bits.

Reply:
I'm not too worried about the test directories as they get cleaned up. But will change